task/manager.go

```diff
@@ -126,6 +131,7 @@ func (m *Manager) Run(ctx context.Context) {
 			default:
 				err := m.cluster.Refresh()
 				if err == nil {
+					m.deleteOrphanCollector()
```

Comment:
**hint**: clean up collectors when the pod no longer exits.  Mainly just to lazily prevent the memory leak.